tools/targets/STM32_gen_PeripheralPins.py

```diff
@@ -1150,40 +1251,90 @@ def parse_BoardFile(fileName):
     for EachPin in PinData:
         try:
             PinLabel[EachPin] = PinData[EachPin]["Signal"]
+            if "LED" in PinLabel[EachPin].upper():
+                LED_list.append(EachPin)
+            elif "BUTTON" in PinLabel[EachPin].upper():
+                BUTTON_list.append(EachPin)
         except:
             pass
 
         try:
             PinLabel[EachPin] = PinData[EachPin]["GPIO_Label"]
+            if "LED" in PinLabel[EachPin].upper():
+                LED_list.append(EachPin)
+            elif "LD1" in PinLabel[EachPin].upper():
+                LED_list.append(EachPin)
+            elif "LD2" in PinLabel[EachPin].upper():
+                LED_list.append(EachPin)
+            elif "LD3" in PinLabel[EachPin].upper():
+                LED_list.append(EachPin)
+            elif "LD4" in PinLabel[EachPin].upper():
+                LED_list.append(EachPin)
+            elif "LD5" in PinLabel[EachPin].upper():
+                LED_list.append(EachPin)
+            elif "LD6" in PinLabel[EachPin].upper():
+                LED_list.append(EachPin)
+            elif "LD7" in PinLabel[EachPin].upper():
+                LED_list.append(EachPin)
+            elif "LD8" in PinLabel[EachPin].upper():
+                LED_list.append(EachPin)
+            elif "LD9" in PinLabel[EachPin].upper():
+                LED_list.append(EachPin)
+            elif "BUTTON" in PinLabel[EachPin].upper():
+                BUTTON_list.append(EachPin)
+            elif "B_USER" in PinLabel[EachPin].upper():
+                BUTTON_list.append(EachPin)
+            elif "BTN" in PinLabel[EachPin].upper():
+                BUTTON_list.append(EachPin)
+        except:
```

Comment:
We shouldn't really use a bare `except` here. I wouldn't expect we want to `pass` on `KeyboardInterrupt`, `SystemExit` or another low level error. A slightly more pleasant version of this would be to catch all errors which derive from `Exception`, but ideally we'd catch the specific exceptions we're expecting here (which I suspect is probably at least a `KeyError' given all the dictionary look-ups).

Reply:
This should be addressed.